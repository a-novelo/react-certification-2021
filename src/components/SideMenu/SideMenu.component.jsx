import React from 'react';
import { useHistory } from 'react-router-dom';
import { SideMenuWrapper, SideMenuOption } from './SideMenu.styled';
import { useAuth } from '../../providers/Auth';

function SideMenu(props) {
  const { show, handleClose } = props;
  const history = useHistory();
  const { authenticated } = useAuth();

  const navigateTo = (page) => {
    handleClose(false);
    history.push(page);
  };

  return (
    <SideMenuWrapper show={show}>
      <SideMenuOption
        onClick={() => {
          navigateTo('/');
        }}
      >
        Home
      </SideMenuOption>
      {authenticated && (
        <SideMenuOption
          onClick={() => {
            navigateTo('/favorites');
          }}
        >
          Favorites
        </SideMenuOption>
      )}
    </SideMenuWrapper>
  );
}

export default SideMenu;
