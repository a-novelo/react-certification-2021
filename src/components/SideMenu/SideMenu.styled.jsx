import styled from 'styled-components';

const SideMenuWrapper = styled.div`
  background-color: white;
  display: ${(props) => (props.show ? 'block' : 'none')};
  position: fixed;
  min-height: 100vh;
  top: 0;
  left: 0;
  width: 20%;
`;

const SideMenuOption = styled.div`
  border-bottom: 1px solid black;
  color: black;
  text-align: initial;
  padding: 10px 20px;
  cursor: pointer;
`;

export { SideMenuWrapper, SideMenuOption };
