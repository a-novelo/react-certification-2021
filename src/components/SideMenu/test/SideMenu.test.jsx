import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import SideMenu from '../SideMenu.component';
import AuthProvider from '../../../providers/Auth';

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('#SideMenu', () => {
  it('displays correctly', () => {
    const tree = renderer
      .create(
        <AuthProvider>
          <SideMenu />
        </AuthProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('goes to the home page', () => {
    render(
      <AuthProvider>
        <SideMenu handleClose={() => {}} />
      </AuthProvider>
    );
    const homeButton = screen.getByText('Home');
    fireEvent(
      homeButton,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      })
    );
  });
});
