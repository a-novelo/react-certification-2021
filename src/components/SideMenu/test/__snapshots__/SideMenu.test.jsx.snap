// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SideMenu displays correctly 1`] = `
<div
  className="sc-bdfBwQ hQCyxS"
>
  <div
    className="sc-gsTCUz faHrza"
    onClick={[Function]}
  >
    Home
  </div>
</div>
`;
